### PR TITLE
feat: Add custom faker

### PR DESCRIPTION
**Pull request by yevgenypats**
https://github.com/bxcodec/faker had a lot of issues
including inherent race conditions and features we didn't need.

This introduces custom faker which will make our life easier
in mock tests.

#### Summary

---

Use the following steps to ensure your PR is ready to be reviewed

- [ ] Read the [contribution guidelines](../blob/main/CONTRIBUTING.md) 🧑‍🎓
- [ ] Run `go fmt` to format your code 🖊
- [ ] Lint your changes via `golangci-lint run` 🚨 (install golangci-lint [here](https://golangci-lint.run/usage/install/#local-installation))
- [ ] Update or add tests 🧪
- [ ] Ensure the status checks below are successful ✅

### CODE REVIEW

File: faker/faker.go [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+
+	if reflectType.Kind() != reflect.Ptr {
+		return fmt.Errorf("object is not a pointer")
+	}
+
+	if reflect.ValueOf(obj).IsNil() {
+		return fmt.Errorf("object is nil %s", reflectType.Elem().String())
+	}
+	f := &faker{
+		max_depth: 12,
```
Comment (erezrokah) [inline faker/faker.go:165]:
```suggestion
		max_depth: 2,
```

I think that should be sufficient for our use cases WDYT?

File: plugins/source_testing.go [hunk 2]
```diff
@@ -26,7 +26,7 @@ func init() {
 // type
 
 func TestSourcePluginSync(t *testing.T, plugin *SourcePlugin, spec specs.Source) {
-	t.Parallel()
+	// t.Parallel()
```
Comment (erezrokah) [inline plugins/source_testing.go:29]:
```suggestion
	// t.Parallel()
```

File: faker/faker.go [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+// 		columnType, err := valueToSchemaType(field.Type)
+// 		if err != nil {
+// 			fmt.Printf("skipping field %s, got err: %v\n", field.Name, err)
+// 			continue
+// 		}
+
+// 	}
+
+// 	return nil
+// }
```
Comment (erezrokah) [inline faker/faker.go:210]:
```suggestion
```

**Review by erezrokah — APPROVED** [review]
Looks good pending some code clean up

File: faker/faker.go (hunk 1 above)
Comment (yevgenypats) [reply faker/faker.go:165]: For me no, usually around 6,7 as I fake the whole response struct and it's quite nested

File: faker/faker.go (hunk 3 above)
Comment (yevgenypats) [reply faker/faker.go:210]: Removed

**Comment by yevgenypats** [comment]
> Looks good pending some code clean up

Thanks! I'll give it some more testing and I think it will be good to merge

**Pull request merged**